src/Stratis.Bitcoin/Controllers/Models/AddressBalancesResult.cs

```diff
@@ -49,4 +50,53 @@ public AddressBalanceResult(string address, Money balance)
         [JsonConverter(typeof(MoneyJsonConverter))]
         public Money Balance { get; set; }
     }
+
+    /// <summary>
+    /// A class that contains a list of balances data per address as requested.
+    /// <para>Should the request fail the <see cref="Reason"/> will be populated.</para>
+    /// </summary>
+    public sealed class VerboseAddressBalancesResult
```

Comment:
Cant you just inherit from `AddressBalancesResult`? There is a lot of duplication here ;)

Reply:
The only duplicated thing there is `public string Reason { get; private set; }` and construction of the failure case using a special constructor. 
Besides the only way to do it properly is to move reason into a base class, otherwise I'll have to unseal `AddressBalancesResult` which will produce even more lines of code. 
So I think there is no need to change anything.